src/offset.js

```diff
@@ -92,7 +92,12 @@ jQuery.fn.extend({
 
 		rect = elem.getBoundingClientRect();
 
-		// Make sure element is not hidden (display: none) or disconnected
+		// Return zeros for disconnected elements (gh-2310)
+		if ( !jQuery.contains( document, elem ) ) {
```

Comment:
I see no reason to treat them differently. Neither have layout, and therefore neither have "real" offset values, yet either could appear in the context collection.